src/listeners/message.ts

```diff
@@ -1,7 +1,28 @@
-import { ActionRowBuilder, ButtonBuilder, ButtonStyle, ChannelType, Client, Colors, EmbedBuilder } from 'discord.js'
+import {
+  ActionRowBuilder,
+  Attachment,
+  ButtonBuilder,
+  ButtonStyle,
+  ChannelType,
+  Client,
+  Collection,
+  Colors,
+  EmbedBuilder
+} from 'discord.js'
 import { ILogger } from '../logger/logger'
 import { containsKeywordFromArray, mentionsBot, greetings, sleepings } from './autoReactHelperFunctions'
 
+function buildAttachmentList(attachments: Collection<string, Attachment>): string {
+  let i = 0
+  const attachmentList = attachments
+    .map((attachment) => {
```

Comment:
you could do map((attachment, index) => ...
then you do not need  let i = 0 and i++ in every iteration

Reply:
The Collections Key and Value is the Attachment ID and the Attachment. So if you do it like this, you will get the Attachment ID instead of an Index. The Attachment ID looks smth like this: 1193265504664698930

Not very suitable imo

Reply:
const attachmentList = attachments
  .map((attachment, index) => `${index + 1}. ${attachment.contentType} - [${attachment.name.substring(0, 30)}](${attachment.url})`)
  .join('\n');

Reply:
this is my suggestion. does the exact same thing like the code before, but uses javascripts functionality of providing an "iterator variable" in .map

Reply:
> this is my suggestion. does the exact same thing like the code before, but uses javascripts functionality of providing an "iterator variable" in .map

Read my previous reply please.

Reply:
[what the actual fuck xD](https://discordjs.guide/additional-info/collections.html#extra-utilities)
you are right. i thought this was just a javascript array, but discord js of course must have their own collection type that has the exact same "streams api" like js arrays. im confused why they dont just use an array but yeah. nevermind

Reply:
> [what the actual fuck xD](https://discordjs.guide/additional-info/collections.html#extra-utilities) you are right. i thought this was just a javascript array, but discord js of course must have their own collection type that has the exact same "streams api" like js arrays. im confused why they dont just use an array but yeah. nevermind

I actually tried to do it like your wrote in the reply, but ended up with an extremely large number and wondered "what the hell?!? shouldn't it be 1, 2, 3?" xD
And tbh i don't really understand why they're doing it like this - i mean you can access the attachment ID via attachment.id...

Reply:
yup, in general the types/classes discord js uses are sooooo weiirrdddd